packages/dialog/src/lit/renderer-directives.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+/* eslint-disable max-classes-per-file */
+import { directive } from 'lit/directive.js';
+import { microTask } from '@vaadin/component-base/src/async.js';
+import { Debouncer } from '@vaadin/component-base/src/debounce.js';
+import { LitRendererDirective } from '@vaadin/lit-renderer';
+
+const CONTENT_UPDATE_DEBOUNCER = Symbol('contentUpdateDebouncer');
+
+class AbstractDialogRendererDirective extends LitRendererDirective {
+  /**
+   * A property to that the renderer callback will be assigned.
+   *
+   * @abstract
+   */
+  rendererProperty;
+
+  /**
+   * Adds the renderer callback to the dialog.
+   */
+  addRenderer() {
+    this.element[this.rendererProperty] = (root, dialog) => {
+      this.renderRenderer(root, dialog);
+    };
+  }
+
+  /**
+   * Runs the renderer callback on the dialog.
+   */
+  runRenderer() {
+    this.element[CONTENT_UPDATE_DEBOUNCER] = Debouncer.debounce(
+      this.element[CONTENT_UPDATE_DEBOUNCER],
+      microTask,
+      () => {
+        this.element.requestContentUpdate();
+      },
+    );
+  }
+
+  /**
+   * Removes the renderer callback from the dialog.
+   */
+  removeRenderer() {
+    this.element[this.rendererProperty] = null;
+  }
+}
+
+export class DialogRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty = 'renderer';
+}
+
+export class DialogHeaderRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty = 'headerRenderer';
+}
+
+export class DialogFooterRendererDirective extends AbstractDialogRendererDirective {
+  rendererProperty = 'footerRenderer';
+}
+
+/**
+ * A Lit directive for populating the content of the dialog.
```

Comment:
Let's extend JSDoc for these functions by adding `@param` annotations - [example](https://github.com/lit/lit/blob/29cc1e6e10d209bbc423e46fbc489b3f8ebc1500/packages/lit-html/src/directives/guard.ts#L81) for `guard()` directive.

Reply:
Done.